test/sqllogictest/chbench.slt

```diff
@@ -446,7 +446,7 @@ ORDER BY o_ol_cnt
 | | implementation = DeltaQuery
 | |   delta %1 %2.(#2, #1, #0)
 | |   delta %2 %1.(#0, #1, #2)
-| Filter (datetots(#4) < 2012-01-02 00:00:00), (datetots(#4) >= 2007-01-02 00:00:00), (#14 >= #4)
+| Filter (datetots(#4) < 2012-01-02 00:00:00), (#14 >= #4), (datetots(#4) >= 2007-01-02 00:00:00)
```

Comment:
I believe these new orderings are correct according to our stated predicate ordering. I worked them out by hand to verify, here are the numbers for this example:

[We first sort on rank_complexity](https://github.com/MaterializeInc/materialize/blob/74db3eb92227cab97517e30178a1e68d6c5c2c09/src/expr/src/relation/canonicalize.rs#L459-L463), and each of these filters has the same rank:

* Both `datetots(#4)...` predicates contain a CallBinary (+1) with a literal and a CallUnary expression (+1) over a Column expression (+1) = rank 3
* The `(#14 >= #4)` predicate is a CallBinary (+1) with two column expressions (+2) = rank 3

Since they have the same rank, we then order these `MirScalarExpr` [by their derived enum order](https://github.com/MaterializeInc/materialize/blob/74db3eb92227cab97517e30178a1e68d6c5c2c09/src/expr/src/scalar/mod.rs#L51-L56).  Since they are all CallBinary types, our secondary sort is their BinaryFunc. [LT comes before GTE](https://github.com/MaterializeInc/materialize/blob/74db3eb92227cab97517e30178a1e68d6c5c2c09/src/expr/src/scalar/func.rs#L2087-L2092), so `(datetots(#4) < 2012-01-02 00:00:00)` is first. 

The next two filters have the same BinaryFunc, so it falls to a tertiary sort on `expr1`. `(#14 > #4)`'s first expression is a `Column`, which comes before the remaining date filter's first expression of type `CallUnary`.